prolog/list_util.pl

```diff
@@ -90,21 +90,9 @@
 %  ?- take(2, L, [a,b]).
 %  L = [a, b|_G1055].
 %  ==
-take(List, N, Front) :-
-    % support deprecated argument order
-    ( integer(N)
-    ; nonvar(List), functor(List,[],0)
-    ; nonvar(List), functor(List,'[|]',2)
-    ),
-    !,
-    print_message(warning, deprecated_argument_order(take)),
-    split_at(N, List, Front, _).
 take(N, List, Front) :-
     split_at(N, List, Front, _).
 
-prolog:message(deprecated_argument_order(Predicate)) -->
-    ['~s(List,N,Front) argument order is deprecated. Use ~s(N,List,Front) instead.'-[Predicate,Predicate]].
-
```

Comment:
Let's have a separate commit for removing the deprecated argument order.  That way `git bisect` can better pinpoint any regressions in causes.